app/scripts/modules/core/pipeline/config/stages/executionWindows/ExecutionWindowActions.tsx

```diff
@@ -0,0 +1,115 @@
+import * as React from 'react';
+
+import {IExecution} from 'core/domain/IExecution';
+import {IExecutionStage} from 'core/domain/IExecutionStage';
+import {ButtonBusyIndicator} from 'core/forms/buttonBusyIndicator/ButtonBusyIndicator';
+import {Application} from 'core/application/application.model';
+import {confirmationModalService} from 'core/confirmationModal/confirmationModal.service';
+import {executionService} from 'core/delivery/service/execution.service';
+import {timePickerTime} from 'core/utils/timeFormatters';
+import {SystemTimezone} from 'core/utils/SystemTimezone';
+import {DAYS_OF_WEEK} from './daysOfWeek';
+
+interface IProps {
+  execution: IExecution;
+  stage: IExecutionStage;
+  application: Application;
+}
+
+interface IState {
+  submitting: boolean;
+}
+
+export class ExecutionWindowActions extends React.Component<IProps, IState> {
+
+  constructor(props: IProps) {
+    super(props);
+    this.state = {
+      submitting: false,
+    };
+  }
+
+  private finishWaiting = () => {
+    const stage = this.props.stage,
+          executionId = this.props.execution.id;
+
+    const matcher = (execution: IExecution) => {
+      const match = execution.stages.find((test) => test.id === stage.id);
+      return match.status !== 'RUNNING';
+    };
+
+    const data = {skipRemainingWait: true};
+    confirmationModalService.confirm({
+      header: 'Really skip execution window?',
+      buttonText: 'Skip',
+      body: '<p>The pipeline will proceed immediately, continuing to the next step in the stage.</p>',
+      submitMethod: () => {
+        this.setState({submitting: true});
+        return executionService.patchExecution(executionId, stage.id, data)
+          .then(() => executionService.waitUntilExecutionMatches(executionId, matcher));
+      }
+    });
+  };
+
+  private replaceDays(days: number[]) {
+    const daySet = new Set(days);
+    return DAYS_OF_WEEK.filter(day => daySet.has(day.ordinal)).map(day => day.label);
+  }
+
+  private getDayText = () => {
+    let dayText = 'Everyday';
+    const days = this.props.stage.context.restrictedExecutionWindow.days;
+    if (days && days.length > 0) {
+      const daysAsText = this.replaceDays(days);
+      dayText = daysAsText.join(', ');
+    }
+    return dayText;
+  };
+
+  public render() {
+    const stage = this.props.stage;
+    const dayText = this.getDayText();
```

Comment:
in theory `this.props.stage.context.restrictedExecutionWindow.days` could change...
but yeah, i don't like having function calls like this in render() then it has to recompute every time the component rerenders anything. Makes sense to have it in state.